code/game/machinery/seed_extractor.dm

```diff
@@ -8,16 +8,53 @@
 
 /obj/structure/machinery/seed_extractor/attackby(obj/item/O as obj, mob/user as mob)
 
+	// Plant bag and other storage containers.
+	if(istype(O,/obj/item/storage))
+		var/obj/item/storage/container = O
+		if(length(container.contents) > 0)
+			to_chat(user, SPAN_NOTICE("You start dumping the contents of [container] into [src]."))
+			if(!do_after(user, 1.5 SECONDS, INTERRUPT_ALL, BUSY_ICON_GENERIC))
+				return
+			for(var/obj/item/item as anything in container)
+				if(!istype(item, /obj/item/reagent_container/food/snacks/grown) && !istype(item, /obj/item/grown))
+					to_chat(user, SPAN_WARNING("Cannot get seeds from [item]."))
+					break
+				else
+					if(user.temp_drop_inv_item(item))
+						var/datum/seed/new_seed_type
+						if(istype(item, /obj/item/grown))
+							var/obj/item/grown/plant = item
+							new_seed_type = GLOB.seed_types[plant.plantname]
+						else
+							var/obj/item/reagent_container/food/snacks/grown/plant = item
+							new_seed_type = GLOB.seed_types[plant.plantname]
+
+						if(new_seed_type)
```

Comment:
same here, just put ! infront like the usual and sort that stuff there. also remove the else clause from this if and put that there.